### ISSUE TITLE
weird characters appear when multi-line defines/error/warning/pragma are used

### ISSUE THREAD

**Issue by YashasSamaga** [Zeex/pawn#191]
```
#warning abcdef \
asdasdas
```

gives

```
warning 237: user warning: abcdef asdasdas
__________________________________^
```
The weird character is the `'\a'` added by `readline` to mark multi-line [insert word]. It uses it internally as a marker but imo it shouldn't end up in any output given to the user (such as #warning).

**Comment by Zeex** [comment]
I think it should work now (fixed in https://github.com/Zeex/pawn/commit/e7ecf386f8e82ed03a3627712d2064940c61aeb4). 

Tested with this code:

```pawn
#warning test
#warning test

#pragma deprecated This function \
is deprecated
native print(const s[]);

// this is a one line comment \
//

#warning my \
                          warning          \
	message

main() {
	print("hello");
	print("world");
}

```

Before:

```
pragma_message_newline.pwn(1) : warning 237: user warning: test
pragma_message_newline.pwn(2) : warning 237: user warning: test
pragma_message_newline.pwn(9) : error 049: invalid line continuation
pragma_message_newline.pwn(13) : warning 237: user warning: my warning          message
pragma_message_newline.pwn(16) : warning 234: function is deprecated (symbol "print") This function is deprecated
pragma_message_newline.pwn(17) : warning 234: function is deprecated (symbol "print") This function is deprecated
```

After:

```
pragma_message_newline.pwn(1) : warning 237: user warning: test
pragma_message_newline.pwn(2) : warning 237: user warning: test
pragma_message_newline.pwn(9) : error 049: invalid line continuation
pragma_message_newline.pwn(13) : warning 237: user warning: my warning message
pragma_message_newline.pwn(16) : warning 234: function is deprecated (symbol "print") This function is deprecated
pragma_message_newline.pwn(17) : warning 234: function is deprecated (symbol "print") This function is deprecated
```

**Comment by YashasSamaga** [comment]
The patch (https://github.com/Southclaws/pawn/commit/693591bdc3491133f2aa90498c4ad8d3aadd0eb7) which was applied for the above fix has ill effects as described in #226.

A new possible solution would be to remove the '\a' character before displaying it to the user instead of removing it before the compilation.
